doc/watermarks_HLD.md

```diff
@@ -0,0 +1,212 @@
+ that period of time. (See open questions)
+
+ Case 4 - Highest watermark over a telemtry period is put into TELEMETRY_WATERMARK table. The maximum is calculated from a series containing the current watermark value(at the end of a period) and the maximum watermark during clearing issued by user (during latest interval).             
+
+# 3 Modules Design
+## 3.1 Modules that need to be updated
+### 3.1.1 CLI
+New script and alias should be implemented to provide watermark values:
+
+$ show priority-group [watermark|persistent-watermark] headroom [-p N]
+```
+Ingress headroom per PG:
+Interface                    PG0   PG1   PG2   PG3   PG4   PG5   PG6   PG7
+Ethernet0                     0     0     0    23     0     0     0     0
+…
+Ethernet128                   0     0     0     0     0     0     0     0
+```
+$ show priority-group [watermark|persistent-watermark] ingress [-p N]
+```
+Ingress shared pool occupancy per PG:
+Interface                   PG0   PG1   PG2   PG3   PG4   PG5   PG6   PG7
+Ethernet0                   0  1092     0   380     0     0     0     0
+…
+Ethernet128                 0     0     0     0     0     0     0     0
+```
+$ show queue [watermark|persistent-watermark] egress unicast [-p N]
+```
+Egress shared pool occupancy per unicast queue:
+Interface                    UC0   UC1   UC2   UC3   UC4   UC5   UC6   UC7
+Ethernet0                      0    14     0    11     0     1     0     0
+…
+Ethernet128                    0     0     0     0     0     0     0     0
+```
+$ show queue [watermark|persistent-watermark] egress multicast [-p N]
+```
+Egress shared pool occupancy per multicast queue:
+Interface                    MC0   MC1   MC2   MC3   MC4   MC5   MC6   MC7
+Ethernet0                      0     3     0     0     0     0     0     0
+…
+Ethernet128                    0     0     0     0     0     0     0     0
+```
+
+In addition clear functionality will be added:
+
+$ clear priority-group [watermark|persistent-watermark] headroom
+
+$ clear priority-group [watermark|persistent-watermark] shared
+
+$ clear queue [watermark|persistent-watermark] unicast
+
+$ clear queue [watermark|persistent-watermark] mutlicast
+
+### 3.1.2 Counter DB
+
+#### The following new Queue counters should be available for each queue entry in the DB:
+- "COUNTERS:queue_vid"
+  - SAI_QUEUE_STAT_SHARED_WATERMARK_BYTES
+#### For every Ingress PG the following should be available in the DB:
+- "COUNTERS:pg_vid"
+  - SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES
+  - SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES
+#### Additionally a few mappings should be added:
+- "COUNTERS_PG_PORT_MAP" - map PG oid to port oid
+- "COUNTERS_PG_NAME_MAP" - map PG oid to PG name
+- "COUNTERS_PG_INDEX_MAP" - map PG oid to PG index
+
+#### New tables will be introduced: 
+ - "TELEMETRY_WATERMARKS" - containing the watermarks for the telemetry to consume
```

Comment:
rename to WATERMARKS

Reply:
need HIGHEST_WATERMARK table to store last highest watermark, this is not persistent watermark table. when user clear the water mark, it needs also clear the highest_watermark.